ingest-app/src/cmr/ingest/services/subscriptions_helper.clj

```diff
@@ -170,12 +173,9 @@
                                         query-params)
               gran-refs (search-gran-refs-by-collection-id context search-by-revision sub-id)
               subscriber-filtered-gran-refs (filter-gran-refs-by-subscriber-id context gran-refs subscriber-id)]]
-    (do
-      (when (seq subscriber-filtered-gran-refs)
-        (send-update-subscription-notification-time! context sub-id))
-      [sub-id subscriber-filtered-gran-refs subscriber-id subscription])))
+    [sub-id subscriber-filtered-gran-refs subscriber-id subscription]))
 
-(defn- send-subscription-emails
+(defn- ^:redef send-subscription-emails
```

Comment:
Why is the `^:redef` here?

Reply:
The desire was to allow more complete testing in the subscription flow.

Previously, notification time is updated after the subscription queries are executed and before emails are sent.  This was to allow testing to run without sending emails.

Ideally, after a subscription email is sent we update the notification time in metadata-db.  Sending emails during testing causes some errors, so I wrote a new function `mock-send-subscription-emails` for test time to override the behavior.  To allow mocking to work using `with-redefs`, the function had to either be set to `^:dynamic` or `^:redef`.  The only documentation I could find for `^:redef` is [here](https://clojure.org/reference/compilation#directlinking). 

"If you wish to mark a var as supporting redefinition (but not dynamic), mark it with ^:redef to avoid direct linking."

Reply:
I missed the whole test change in the review, so I missed `with-redefs` reference. I have put my comment on the test itself.